### PR TITLE
Refs #30365: Ignore symlinks for puppet executable

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ekohl** [comment]
Hmmm, tests are not happy

**Comment by ehelms** [comment]
Took some finagling, @ekohl do check the test updates for the structural changes

File: kafo_parsers.gemspec [hunk 1]
```diff
@@ -16,11 +16,12 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ["lib"]
 
-  spec.add_development_dependency "bundler", "~> 1.5"
+  spec.add_development_dependency "bundler"
   spec.add_development_dependency "rake"
   spec.add_development_dependency "minitest"
   spec.add_development_dependency "simplecov"
   spec.add_development_dependency "ci_reporter_minitest"
+  spec.add_development_dependency "mocha"
```
Comment (ekohl) [inline kafo_parsers.gemspec:24]: I'd like to avoid adding another way to mock.

**Comment by ehelms** [comment]
@ekohl this needs a corresponding look

**Comment by ehelms** [comment]
IIRC when I was building the fix, this is cached at a class level which made it hard to test given in tests the paths needed to change to test different combinations. The call to find the bin file did not look expensive so I chose testing over performance gains from caching.

**Comment by ekohl** [comment]
You can split the logic into two methods: `find_puppet_bin` and `puppet_bin` where the latter only performs the caching.

**Comment by ehelms** [comment]
@ekohl Logic is now split

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**